src/main/java/com/powsybl/openloadflow/sa/AcSecurityAnalysis.java

```diff
@@ -135,10 +138,14 @@ private SecurityAnalysisResult runSimulations(LfNetwork network, List<Propagated
             if (preContingencyComputationOk) {
                 detectViolations(network.getBranches().stream(), network.getBuses().stream(), preContingencyLimitViolations);
 
-                LOGGER.info("Save pre-contingency state");
+                // update voltage and active power targets to saved state
+                networkState.getBusStates().forEach(busState -> {
+                    busState.updateVoltage();
+                    busState.updateTargetP();
```

Comment:
Yes I get you! Here you re-save specific fields, that is why you call that `update` ! So for me, it is a good design but:
- If some bus where PV and have switch PQ, you have to re-start from the PQ state. All is design for this bus to be able to switch back PV after a contingency. The fields that you need to store are numerous.
- If you distribute the slack on loads keeping power factor constant, you also have to save the `loadTargetQ`.
- After the pre-contingency computation, all the discrete regulation are off. I think that we have to restore them, in order to be able to re-run them after a contingency. It could be optional.